packages/cloudflare/src/cli/templates/images.ts

```diff
@@ -83,6 +96,21 @@ export function matchRemotePattern(pattern: RemotePattern, url: URL): boolean {
   return true;
 }
 
+export function matchLocalPattern(pattern: LocalPattern, url: URL): boolean {
+  // https://github.com/vercel/next.js/blob/d76f0b1/packages/next/src/shared/lib/match-local-pattern.ts
+  if (pattern.search !== undefined) {
+    if (pattern.search !== url.search) {
+      return false;
+    }
+  }
```

Comment:
nit: I removed the unneeded nested if in remote match too
```suggestion
  if (pattern.search !== undefined && pattern.search !== url.search) {
    return false;
  }
```